gadget/layout.go

```diff
@@ -76,10 +76,12 @@ type PartiallyLaidOutVolume struct {
 	LaidOutStructure []LaidOutStructure
 }
 
-// LaidOutStructure describes a VolumeStructure that has been placed within the
-// volume
+// LaidOutStructure describes an OnDiskStructure that has been placed within the
+// volume.
 type LaidOutStructure struct {
-	*VolumeStructure
+	OnDiskStructure
+	// GadgetStructure is the volume structure defined in gadget.yaml
+	GadgetStructure *VolumeStructure
```

Comment:
wouldn't the diff be smaller if we kept calling this VolumeStructure? was there some issue with that? I agree this is a better name but we could have a later more mechanical change VolumeStructure -> GadgetStructure  here, and OnDiskStructure -> StorageStructure